PWGDQ/Tasks/dqEfficiency.cxx

```diff
@@ -835,13 +836,210 @@ struct AnalysisSameEventPairing {
+    VarManager::SetDefaultVarNames();
+    fHistMan = new HistogramManager("analysisHistos", "aa", VarManager::kNVars);
+    fHistMan->SetUseDefaultVariableNames(kTRUE);
+    fHistMan->SetDefaultVarNames(VarManager::fgVariableNames, VarManager::fgVariableUnits);
+
+    // TODO: Create separate histogram directories for each selection used in the creation of the dileptons
+    // TODO: Implement possibly multiple selections for the associated track ?
+    if (context.mOptions.get<bool>("processDimuonMuonSkimmed")) {
+      //DefineHistograms(fHistMan, "DileptonsSelected;DileptonTrackInvMass;DileptonsSelected_matchedMC;DileptonTrackInvMass_matchedMC;"); // define all histograms
+      //VarManager::SetUseVars(fHistMan->GetUsedVars());
+      //fOutputList.setObject(fHistMan->GetMainHistogramList());
+
+      histNames += "DileptonsSelected;DileptonTrackInvMass;";
+
+      if (!sigNamesStr.IsNull()) {
+        for (int isig = 0; isig < objRecSigArray->GetEntries(); ++isig) {
+          MCSignal* sig = o2::aod::dqmcsignals::GetMCSignal(objRecSigArray->At(isig)->GetName());
+          if (sig) {
+            if (sig->GetNProngs() == 1) {
+              fRecMCSignals.push_back(*sig);
+              TString histName = Form("LeptonsSelected_matchedMC_%s", sig->GetName());
+              histNames += Form("%s;", histName.Data());
+              fRecMCSignalsNames.push_back(sig->GetName());
+            }
+            if (sig->GetNProngs() == 2) {
+              fRecMCSignals.push_back(*sig);
+              TString histName = Form("DileptonsSelected_matchedMC_%s", sig->GetName());
+              histNames += Form("%s;", histName.Data());
+              fRecMCSignalsNames.push_back(sig->GetName());
+            }
+            if (sig->GetNProngs() == 3) {
+              fRecMCSignals.push_back(*sig);
+              TString histName = Form("DileptonTrackInvMass_matchedMC_%s", sig->GetName());
+              histNames += Form("%s;", histName.Data());
+              fRecMCSignalsNames.push_back(sig->GetName());
+            }
+          }
+        }
+      }
+
+      DefineHistograms(fHistMan, histNames.Data()); // define all histograms
+      VarManager::SetUseVars(fHistMan->GetUsedVars());
+      fOutputList.setObject(fHistMan->GetMainHistogramList());
+    }
+
+    TString configCutNamesStr = fConfigTrackCuts.value;
+    if (!configCutNamesStr.IsNull()) {
+      std::unique_ptr<TObjArray> objArray(configCutNamesStr.Tokenize(","));
+      fNHadronCutBit = objArray->GetEntries();
+    } else {
+      fNHadronCutBit = 0;
+    }
+  }
+
+  // Template function to run pair - track combinations
+  template <int TPairType, uint32_t TEventFillMap, uint32_t TEventMCFillMap, uint32_t TTrackFillMap, typename TEvent, typename TTracks, typename TEventsMC, typename TTracksMC>
+  void runDileptonLepton(TEvent const& event, TTracks const& tracks, soa::Join<aod::Dileptons, aod::DileptonsExtra> const& dileptons, TEventsMC const& eventsMC, TTracksMC const& tracksMC)
```

Comment:
Since for now there is nothing specific here, i would call this function "runDileptonTrack" since ideally we will use it for most 3 prong cases.

Reply:
Done!